packages/core/src/render3/di.ts

```diff
@@ -317,18 +317,18 @@ class ElementRef implements viewEngine.ElementRef {
  */
 export function getOrCreateTemplateRef<T>(di: LNodeInjector): viewEngine.TemplateRef<T> {
   const data = (di.node as LContainer).data;
-  if (data === null || data.template === null) {
+  /*if (data === null || data.template === null) {
     throw createInjectionError('Directive does not have a template.', null);
-  }
+  }*/
   return di.templateRef ||
       (di.templateRef = new TemplateRef<any>(getOrCreateElementRef(di), data.template));
 }
 
 /** A ref to a particular template. */
-class TemplateRef<T> implements viewEngine.TemplateRef<T> {
+export class TemplateRef<T> implements viewEngine.TemplateRef<T> {
```

Comment:
You should not need to export these. In tests just check for shape or constructor name.